foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/mutating/webhook/ConfigMutatingWebhook.java

```diff
@@ -33,31 +48,124 @@
 @Path("/webhook")
 @ApplicationScoped
 public class ConfigMutatingWebhook {
+    public static final String KRAUSENING_PASSWORD = "KRAUSENING_PASSWORD";
+    public static final String CONFIG_STORE_INJECT_START = "\\$getConfigValue\\(";
+    public static final String CONFIG_STORE_INJECT_END = ")";
+    public static final String PROPERTY_KEY_DELIMITER = ";";
+    public static final String GROUP_NAME = "groupName";
+    public static final String PROPERTY_NAME = "propertyName";
     private static final Logger logger = LoggerFactory.getLogger(ConfigMutatingWebhook.class);
+
+
+    @Inject
+    public ConfigLoader configLoader;
     
     @POST
     @Path("/process")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     public AdmissionReview validate(AdmissionReview admissionReviewRequest) {
-        logger.info("Webhook was run!");
-        logger.info("Request: {}", admissionReviewRequest.getRequest().getKind().toString());
-        
-        // Currently the response does contain any patches, resulting in the requested resource remaining the same
-        // TODO: create a patch for injecting values into the yaml
+        String masterPassword = System.getProperty(KRAUSENING_PASSWORD);
+        if (StringUtils.isBlank(masterPassword) && StringUtils.isBlank(System.getenv(KRAUSENING_PASSWORD))) {
+            logger.warn("No {} set, the encryped property value will remain encrypted!", KRAUSENING_PASSWORD);
+        }
+        logger.info("Webhook intercepts kubernetes resource request: [name: {}, kind: {}]", admissionReviewRequest.getRequest().getName(), admissionReviewRequest.getRequest().getKind().getKind());
+
         AdmissionRequest request = admissionReviewRequest.getRequest();
 
-        AdmissionResponse response = new AdmissionResponseBuilder()
-                                                    .withAllowed(true)
-                                                    .withUid(request.getUid())
-                                                    .build();
+        JsonFactory factory = JsonFactory.builder()
+                .enable(JsonReadFeature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER)
+                .enable(JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS)
+                .enable(JsonReadFeature.ALLOW_JAVA_COMMENTS)
+                .enable(JsonReadFeature.ALLOW_YAML_COMMENTS)
+                .enable(JsonReadFeature.ALLOW_SINGLE_QUOTES)
+                .build();
+
+        ObjectMapper mapper = new ObjectMapper(factory);
+        AdmissionResponse response;
+        try {
+            JsonNode requestJson = mapper.readTree(new StringReader(mapper.writeValueAsString(request)));
```

Comment:
S: This is really odd to me. Why are we writing the request to a string, then to a generic JSON object, then back to a string? Why not just do:
```
String source = mapper.wratieValueAsString(request.getObject());
StringBuilder target = replaceConfigStorKeyWithConfigValue(source);
```

Reply:
yeah that makes sense. Let me simplify it and parse to Json later for comparison.